internal/provider/resource/project_user_resource.go

```diff
@@ -345,11 +346,19 @@ func (r *ProjectUserResource) Read(ctx context.Context, req resource.ReadRequest
 		ProjectID: state.ProjectID.ValueString(),
 		Username:  state.Username.ValueString(),
 	})
+
 	if err != nil {
+		// If the membership isn't found for whatever reason (user left the project, etc), remove the resource so we can re-create it, and fix the state.
+		if strings.Contains(err.Error(), "Project membership not found") {
```

Comment:
We should make use of the `ErrNotFound` error instead of this string based apprach